### PR TITLE
feat(donasi): Donasi Warga page + widget

### LINKED ISSUE #329: Donasi Warga page + widget
## UI Design

image

image

## Implementation Model

To make this implementation quick, let's implement the widget with dummy data from JS first, before converting said dummy data into the final Netlify CMS collection.

## Tasks

- [ ] Create Donasi Warga widget on homepage with pulling dummy data from JS
- [ ] Create Donasi Warga page using said dummy data
- [ ] Convert JS dummy data into Netlify CMS collection
- [ ] Update page + widget to pull data from new Netlify CMS collection

### PR DESCRIPTION
**Pull request by syauqy**
Closes #329  

## Description

image

- the `donasi.tsx` page added 
- `donation-card` reusable component added .
- `donasi_page` collection on `config.yml` added 
- `informasi_donasi.json` contents added 

## Current Tasks

- [x] create a donasi page
- [x] create a donasi widget on the homepage
- [x] add donasi warga collections on `config.yml`
- [x] add donasi warga page contents
- [x] write the integration test for donasi page and donasi widget

### CODE REVIEW

**Comment by syauqy** [comment]
donasi widget added.

image

File: pages/donasi.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+};
+
+const meta = {
+  title: `Donasi dan Penggalangan Dana`,
+  description: `Kumpulan informasi mengenai donasi, nomor rekening, dan lainnya yang relevan dengan bantuan COVID-19.`,
+};
+
+export default function DonasiPage(props: InformasiDonasiProps) {
+  return (
+    <div>
```
Comment (resir014) [inline pages/donasi.tsx:27]:
```suggestion
```

Don't wrap `Page` inside another div.

File: pages/donasi.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                      url={donation.url}
+                    />
+                  ),
+                )}
+              </div>
+            </div>
+          </Container>
+        </InternalPageContent>
+      </Page>
+    </div>
```
Comment (resir014) [inline pages/donasi.tsx:70]:
```suggestion
```

Idem.

File: pages/donasi.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                  href: "/donasi",
+                  current: true,
+                },
+              ]}
+              title="Donasi dan Penggalangan Dana"
+            />
+            <div className="p-4 overflow-hidden rounded-md space-y-6 bg-white">
+              <div>
+                <div className="text-sm text-gray-600 mb-4">
+                  {meta.description}
```
Comment (resir014) [inline pages/donasi.tsx:50]: Put this on the `description` prop on `PageHeader` instead. This is added on #574.

**Review by resir014 — CHANGES_REQUESTED** [review]
Some changes necessary, please take a look. @syauqy

File: pages/donasi.tsx (hunk 3 above)
Comment (syauqy) [reply pages/donasi.tsx]: ahh got it. Thanks! 🙇

**Review by mazipan — APPROVED** [review]
Looks good to me.

Just a question, do we need to add "donasi" on the hamburger menu?

**Comment by resir014** [comment]
@mazipan Yes. We need to add that to `~/lib/layout/navigation-data.tsx`. But I think we can do that on a separate PR as per usual.

**Comment by mazipan** [comment]
Then, it's your call @resir014 
All good from me.

Already check the preview page, looks good to me.

**Review by resir014 — APPROVED** [review]

File: public/admin/config.yml [hunk 4]
```diff
@@ -243,3 +246,37 @@ collections:
...
                   widget: "string"
                 }
+  - name: "donasi_page"
+    label: "Donasi Page"
+    delete: false
+    preview_path: "donasi"
+    files:
+      - name: "informasi_donasi"
+        label: "Informasi Donasi"
+        file: "_content/donasi/informasi-donasi.json"
```
Comment (zainfathoni) [inline public/admin/config.yml:256]:
```suggestion
        file: "_content/donasi.json"
```

Please simplify the file structure, thanks!

File: lib/content/informasi-donasi.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import informasiDonasi from "~/_content/donasi/informasi-donasi.json";
```
Comment (zainfathoni) [inline lib/content/informasi-donasi.ts:1]:
```suggestion
import donasi from "~/_content/donasi.json";
```

Please simplify the file structure, thanks!

**Review by zainfathoni — DISMISSED** [review]
Sorry for nitpicking, but I don't think we need a folder here. 😅 

Please make the changes accordingly. Thanks, @syauqy! 🙏

**Review by mazipan — APPROVED** [review]

**Comment by syauqy** [comment]
> Sorry for nitpicking, but I don't think we need a folder here. 😅
> 
> Please make the changes accordingly. Thanks, @syauqy! 🙏

😄  sure mas @zainfathoni, thank you for the suggestion

**Pull request merged**